crates/boundless-market/src/contracts/boundless_market.rs

```diff
@@ -1274,16 +1287,54 @@ where
     }
 }
 
+fn decode_calldata(data: &Bytes) -> Result<Vec<Fulfillment>> {
+    if let Ok(call) = IBoundlessMarket::submitRootAndFulfillBatchCall::abi_decode(data, true) {
+        return Ok(call.fills);
+    }
+    if let Ok(call) =
+        IBoundlessMarket::submitRootAndFulfillBatchAndWithdrawCall::abi_decode(data, true)
+    {
+        return Ok(call.fills);
+    }
+    if let Ok(call) = IBoundlessMarket::fulfillCall::abi_decode(data, true) {
+        return Ok(vec![call.fill]);
+    }
+    if let Ok(call) = IBoundlessMarket::fulfillBatchCall::abi_decode(data, true) {
+        return Ok(call.fills);
+    }
+    if let Ok(call) = IBoundlessMarket::fulfillAndWithdrawCall::abi_decode(data, true) {
+        return Ok(vec![call.fill]);
+    }
+    if let Ok(call) = IBoundlessMarket::fulfillBatchAndWithdrawCall::abi_decode(data, true) {
+        return Ok(call.fills);
+    }
+    if let Ok(call) = IBoundlessMarket::priceAndFulfillCall::abi_decode(data, true) {
+        return Ok(vec![call.fill]);
+    }
+    if let Ok(call) = IBoundlessMarket::priceAndFulfillBatchCall::abi_decode(data, true) {
+        return Ok(call.fills);
+    }
+    if let Ok(call) = IBoundlessMarket::priceAndFulfillAndWithdrawCall::abi_decode(data, true) {
+        return Ok(vec![call.fill]);
+    }
+    if let Ok(call) = IBoundlessMarket::priceAndFulfillBatchAndWithdrawCall::abi_decode(data, true)
+    {
+        return Ok(call.fills);
+    }
+
+    Err(anyhow!("Failed to decode calldata as any fulfillment call"))
```

Comment:
We should include the hex-encoded selector (i.e. first 4 bytes of calldata) or the full calldata here to help with debugging if this error occurs. It would definitely be easy to add a new fulfill entrypoint without remembering to update this list.